src/routes/handlers/kafkaAsyncGoldenIdUpdater.ts

```diff
@@ -0,0 +1,96 @@
+import { Kafka, logLevel } from 'kafkajs';
+
+import { getConfig } from '../../config/config';
+import logger from '../../logger';
+import { JempiAudit } from '../../types/jempiAudit';
+import { fetchMpiResourceByRef } from '../../utils/mpi';
+import { Patient } from 'fhir/r3';
+
+export const asyncGoldenIdUpdater = async (): Promise<void> => {
+  const config = getConfig();
+
+  const kafka = new Kafka({
+    logLevel: logLevel.ERROR,
+    clientId: config.mpiKafkaClientId,
+    brokers: config.kafkaBrokers.split(','),
+  });
+
+  const consumer = kafka.consumer({ groupId: 'mpi-mediator-golden-id' });
+  const producer = kafka.producer();
+
+  await consumer.connect();
+  await consumer.subscribe({
+    topic: config.kafkaJempiAuditTopic,
+  });
+
+  await producer.connect();
+
+  logger.info('Kafka golden ID consumer started');
+
+  await consumer.run({
+    eachMessage: async ({ message }) => {
+      logger.info('JeMPI audit received from queue');
+
+      if (message.value === null) {
+        logger.info('JeMPI audit message is null');
+
+        return;
+      }
+
+      const audit: JempiAudit = JSON.parse(message.value.toString());
+
```

Comment:
The app will crash if a message has invalid json. Maybe we should put this in a try/catch block, and log the error

Reply:
Good idea, will do.